hatchery/src/snapshot.rs

```diff
@@ -0,0 +1,138 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use crate::error::Error;
+use crate::storage_helpers::{
+    combine_module_snapshot_names, snapshot_id_to_name,
+};
+use crate::Error::PersistenceError;
+use std::io::Read;
+use std::path::{Path, PathBuf};
+
+use rkyv::{Archive, Deserialize, Serialize};
+pub const SNAPSHOT_ID_BYTES: usize = 32;
+#[derive(
+    Debug,
+    PartialEq,
+    Eq,
+    Archive,
+    Serialize,
+    Deserialize,
+    PartialOrd,
+    Ord,
+    Hash,
+    Clone,
+    Copy,
+)]
+#[repr(C)]
+pub struct SnapshotId([u8; SNAPSHOT_ID_BYTES]);
+impl SnapshotId {
+    pub const fn uninitialized() -> Self {
+        SnapshotId([0u8; SNAPSHOT_ID_BYTES])
+    }
+    pub fn as_bytes(&self) -> &[u8] {
+        &self.0
+    }
+}
```

Comment:
Are we thinking that we might pass a snapshot to a contract? If not, I would say it doesn't need to be `repr(C)` and `uninitialized` might not make sense.

Reply:
we will need to store it, but point taken